cpp/tests/groupby/group_max_scan_test.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tests/groupby/groupby_test_util.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/dictionary/update_keys.hpp>
+
+namespace cudf {
+namespace test {
+template <typename V>
+struct groupby_max_scan_test : public cudf::test::BaseFixture {
+};
+
+TYPED_TEST_CASE(groupby_max_scan_test, cudf::test::FixedWidthTypesWithoutFixedPoint);
+
+TYPED_TEST(groupby_max_scan_test, basic)
+{
+  using K = int32_t;
+  using V = TypeParam;
+  using R = cudf::detail::target_type_t<V, aggregation::MAX>;
+
+  // clang-format off
+  fixed_width_column_wrapper<K> keys          {1, 2, 3, 1, 2, 2, 1, 3, 3, 2};
+  fixed_width_column_wrapper<V, int32_t> vals({5, 6, 7, 8, 9, 0, 1, 2, 3, 4});
+
+  fixed_width_column_wrapper<K> expect_keys          {1, 1, 1, 2, 2, 2, 2, 3, 3, 3};
+                                                  // {5, 8, 1, 6, 9, 0, 4, 7, 2, 3}
+  fixed_width_column_wrapper<R, int32_t> expect_vals({5, 8, 8, 6, 9, 9, 9, 7, 7, 7});
```

Comment:
Similarly, aliasing for long names such as `fixed_width_column_wrapper<K>` will make the code looks cleaner.